### PR TITLE
fix bug in freeVars code

### PR DESCRIPTION
**Pull request by nickchapman-da**
fix bug spotted in passing

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Type.hs [hunk 1]
```diff
@@ -32,7 +32,7 @@ freeVars e = go Set.empty e Set.empty
         TBuiltin _ -> acc
         TForall (v, _k) s -> go (Set.insert v boundVars) s acc
         TStruct fs -> foldl' (\acc (_, t) -> go boundVars t acc) acc fs
-        TNat _ -> Set.empty
+        TNat _ -> acc
```
Comment (hurryabit) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Type.hs]: The argument of `TNat` can contain type variables as well, can't it?

File: compiler/daml-lf-ast/test/DA/Daml/LF/Ast/Tests.hs [hunk 2]
```diff
@@ -63,6 +64,17 @@ substitutionTests = testGroup "substitution"
                 TBuiltin BTArrow `TApp` vBeta11 `TApp` vBeta1
             ty2 = substitute subst ty1
         assertBool "wrong substitution" (alphaEquiv ty1 ty2)
+
+    , testCase "freeVars/TypeLevelNat" $ do
+        let -- no thank you HLint
```
Comment (hurryabit) [inline compiler/daml-lf-ast/test/DA/Daml/LF/Ast/Tests.hs]: What is this about?

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Type.hs (hunk 1 above)
Comment (associahedron) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Type.hs]: It contains only a `TypeLevelNat`, which is a newtype over `Int`, so no.

Comment (hurryabit) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Type.hs]: Oh right, I confused it with `TNumeric`. My bad.

File: compiler/daml-lf-ast/test/DA/Daml/LF/Ast/Tests.hs (hunk 2 above)
Comment (nickchapman-da) [reply compiler/daml-lf-ast/test/DA/Daml/LF/Ast/Tests.hs]: Adding a lone `let` prevents HLint complaining about the unnecessary parens, which were added in the testcase to improve layout. But I removed these parens now.

Comment (cocreature) [reply compiler/daml-lf-ast/test/DA/Daml/LF/Ast/Tests.hs]: thanks!

**Pull request merged**